project/Dist.scala

```diff
@@ -6,62 +6,58 @@ package akka
+  val distUnzipped = SettingKey[File]("dist-unzipped")
+  val distFile = SettingKey[File]("dist-file")
+
   val distExclude = SettingKey[Seq[String]]("dist-exclude")
   val distAllClasspaths = TaskKey[Seq[Classpath]]("dist-all-classpaths")
   val distDependencies = TaskKey[Seq[File]]("dist-dependencies")
   val distLibJars = TaskKey[Seq[File]]("dist-lib-jars")
   val distSrcJars = TaskKey[Seq[File]]("dist-src-jars")
   val distDocJars = TaskKey[Seq[File]]("dist-doc-jars")
   val distSources = TaskKey[DistSources]("dist-sources")
-  val distDirectory = SettingKey[File]("dist-directory")
-  val distUnzipped = SettingKey[File]("dist-unzipped")
-  val distFile = SettingKey[File]("dist-file")
   val dist = TaskKey[File]("dist", "Create a zipped distribution of everything.")
 
   lazy val settings: Seq[Setting[_]] = Seq(
-    distExclude := Seq.empty,
-    distAllClasspaths <<= (thisProjectRef, buildStructure, distExclude) flatMap aggregated(dependencyClasspath.task in Compile),
+    distAllClasspaths <<= (thisProjectRef, buildStructure) flatMap aggregated(dependencyClasspath in Compile),
     distDependencies <<= distAllClasspaths map { _.flatten.map(_.data).filter(ClasspathUtilities.isArchive).distinct },
-    distLibJars <<= (thisProjectRef, buildStructure, distExclude) flatMap aggregated(packageBin.task in Compile),
-    distSrcJars <<= (thisProjectRef, buildStructure, distExclude) flatMap aggregated(packageSrc.task in Compile),
-    distDocJars <<= (thisProjectRef, buildStructure, distExclude) flatMap aggregated(packageDoc.task in Compile),
+    distLibJars <<= (thisProjectRef, buildStructure) flatMap aggregated(packageBin in Compile),
+    distSrcJars <<= (thisProjectRef, buildStructure) flatMap aggregated(packageSrc in Compile),
+    distDocJars <<= (thisProjectRef, buildStructure) flatMap aggregated(packageDoc in Compile),
     distSources <<= (distDependencies, distLibJars,  distSrcJars, distDocJars, doc in ScalaUnidoc, generate in Sphinx in docsProject) map DistSources,
     distDirectory <<= crossTarget / "dist",
     distUnzipped <<= distDirectory / "unzipped",
-    distFile <<= (distDirectory, version, scalaBinaryVersion) { (dir, v, sbv) => 
+    distFile <<= (distDirectory, version, scalaBinaryVersion) { (dir, v, sbv) =>
       dir / ("akka_" + sbv + "-" + v + ".zip") },
     dist <<= distTask
   )
 
   def docsProject: ProjectReference = LocalProject(AkkaBuild.docs.id)
 
-  def aggregated[T](task: SettingKey[Task[T]])(projectRef: ProjectRef, structure: Load.BuildStructure, exclude: Seq[String]): Task[Seq[T]] = {
-    val projects = aggregatedProjects(projectRef, structure, exclude)
-    projects flatMap { task in LocalProject(_) get structure.data } join
-  }
+  def aggregated[T](task: TaskKey[T])(projectRef: ProjectRef, structure: BuildStructure): Task[Seq[T]] =
+    aggregatedProjects(projectRef, structure)
+      .filter { publishArtifact in _ get structure.data getOrElse false }
```

Comment:
Aggregated projects now get included even if their parents have `publishArtifact := false`. Dunno if that's a problem.

I looked into `ScopeFilter` but only the predefined project filters seems to work. :(